packages/coding-standard/src/Rules/CheckRequiredInterfaceInContractNamespaceRule.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Rules;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Node\Stmt\UseUse;
+use PHPStan\Analyser\Scope;
+use Symplify\CodingStandard\ValueObject\PHPStanAttributeKey;
+
+/**
+ * @see \Symplify\CodingStandard\Tests\Rules\CheckRequiredInterfaceInContractNamespaceRule\CheckRequiredInterfaceInContractNamespaceRuleTest
+ */
+final class CheckRequiredInterfaceInContractNamespaceRule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Interface is required in Contract namespace';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/kmrIG1/1
+     */
+    private const A_CONTRACT_NAMESPACE_REGEX = '#\bContract\b#';
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [UseUse::class];
+    }
+
+    /**
+     * @param UseUse $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        $name = (string) $node->name;
+        if (interface_exists($name)) {
+            return [];
+        }
+
+        $namespace = $node->getAttribute(PHPStanAttributeKey::PARENT);
+        while ($namespace) {
+            if ($namespace instanceof Namespace_) {
+                break;
+            }
+
+            $namespace = $namespace->getAttribute(PHPStanAttributeKey::PARENT);
+        }
```

Comment:
It might be more clear to use `$scope->getClassReflection()` and get namespace name from it.

The `getNodeTypes()` could listen to `Interface_` type.


---


I might have failed at providing clear issue description. I'll try to fix it examples:


```php
namespace Whatever;

class SomeClass
{
}
```

Don't care :heavy_check_mark: 

<br>


```php
namespace Whatever\Contract;

interface Anything
{
}
```

It is the `interface` and in `Contract` namespace - good :heavy_check_mark: 


<br>


```php
namespace Whatever\Anything;

interface Anything
{
}
```

It is the `interface` and **not in `Contract` namespace** - error :no_entry: :

Reply:
@TomasVotruba `$scope->getClassReflection()` on `UseUse` got null, not `ClassReflection`, I tried that.

Reply:
Ok, I will try use `Interface_` node types

Reply:
@TomasVotruba I tried using `Interface_` for node types, and `$scope->getClassReflection()` got null.

Reply:
implemented with `Interface_` statement, getting namespace is a parent of the `Interface_` itself.